functions/go/apply-setters/apply_setters.go

```diff
@@ -0,0 +1,241 @@
+package main
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"sigs.k8s.io/kustomize/kyaml/errors"
+	"sigs.k8s.io/kustomize/kyaml/kio"
+	"sigs.k8s.io/kustomize/kyaml/yaml"
+)
+
+var _ kio.Filter = ApplySetters{}
+
+// ApplySetters applies the setter values to the resource fields which are tagged
+// by the setter reference comments
+type ApplySetters struct {
+	// Setters holds the user provided values for all the setters
+	Setters []Setter `json:"setters,omitempty" yaml:"setters,omitempty"`
+}
+
+type Setter struct {
+	// Name is the name of the setter
+	Name string `json:"name,omitempty" yaml:"name,omitempty"`
+
+	// Value is the input value for setter
+	Value string `json:"value,omitempty" yaml:"value,omitempty"`
+}
+
+const SetterCommentIdentifier = "# kpt-set: "
+
+// Filter implements Set as a yaml.Filter
+func (as ApplySetters) Filter(nodes []*yaml.RNode) ([]*yaml.RNode, error) {
+	for i := range nodes {
+		err := accept(&as, nodes[i])
+		if err != nil {
+			return nil, errors.Wrap(err)
+		}
+	}
+	return nodes, nil
+}
+
+// visitMapping takes input mapping node, and performs following steps
+// checks if the key node of the input mapping node has line comment with SetterCommentIdentifier
+// checks if the value node is of sequence node type
+// if yes to both, resolves the setter value for the setter name in the line comment
+// replaces the existing sequence node with the new values provided by user
+func (as *ApplySetters) visitMapping(object *yaml.RNode) error {
+	return object.VisitFields(func(node *yaml.MapNode) error {
+		if node == nil || node.Key == nil || node.Key.YNode() == nil ||
```

Comment:
There's probably a utlity for this common check already (IsNilOrEmpty does a subset of this). Try not to roll your own.